system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -157,76 +149,25 @@ export class TopSite extends React.PureComponent {
   onMenuUpdate(showContextMenu) {
     this.setState({showContextMenu});
   }
-  onDismissButtonClick() {
-    const {link} = this.props;
-    if (link.isPinned) {
-      this.props.dispatch(ac.SendToMain({
-        type: at.TOP_SITES_UNPIN,
```

Comment:
Odd this was here to begin with as blocking will unpin https://searchfox.org/mozilla-central/source/toolkit/modules/NewTabUtils.jsm#556